damus/Views/Events/FollowPack/FollowPackPreview.swift

```diff
@@ -0,0 +1,215 @@
+                switch displayName {
+                case .one(let one):
+                    Text(one)
+                        .font(.subheadline).foregroundColor(.gray)
+                    
+                case .both(username: let username, displayName: let displayName):
+                        HStack(spacing: 6) {
+                            Text(verbatim: displayName)
+                                .font(.subheadline).foregroundColor(.gray)
+                            
+                            Text(verbatim: "@\(username)")
+                                .font(.subheadline).foregroundColor(.gray)
+                        }
+                }
+            }
+            .padding(.horizontal, 10)
+            
+            HStack(alignment: .center) {
+                if !event.publicKeys.isEmpty {
+                    CondensedProfilePicturesView(state: state, pubkeys: event.publicKeys, maxPictures: 5)
+                    
+                    Text("· \(event.publicKeys.count) users")
+                        .font(.subheadline).foregroundColor(.gray)
+                        .padding(.top, 4)
+                } else {
+                    Text("0 users")
+                        .font(.subheadline).foregroundColor(.gray)
+                        .multilineTextAlignment(.leading)
+                        .padding(.top, 7)
+                }
+            }
+            .padding(.horizontal, 10)
+            .padding(.bottom, 20)
+        }
+        .frame(width: 350, height: 300, alignment: .leading)
+        .background(DamusColors.neutral3)
+        .cornerRadius(10)
+        .overlay(
+            RoundedRectangle(cornerRadius: 10)
+                .stroke(DamusColors.neutral1, lineWidth: 1)
+        )
+        .padding(.top, 10)
+        .onAppear {
+            blur_images = should_blur_images(settings: state.settings, contacts: state.contacts, ev: event.event, our_pubkey: state.pubkey)
+        }
+    }
+}
+
+struct FollowPackPreview: View {
+    let state: DamusState
+    let event: FollowPackEvent
+    let options: EventViewOptions
+
+    init(state: DamusState, ev: NostrEvent, options: EventViewOptions) {
+        self.state = state
+        self.event = FollowPackEvent.parse(from: ev)
+        self.options = options.union(.no_mentions)
+    }
+
+    var body: some View {
+        FollowPackPreviewBody(state: state, ev: event, options: options, header: false)
+    }
+}
+
+let test_follow_list_event = FollowPackEvent.parse(from: NostrEvent(
+    content: "",
+    keypair: test_keypair,
+    kind: NostrKind.longform.rawValue,
+    tags: [
+        ["title", "DAMUSES"],
+        ["description", "Damus Team"],
+        ["published_at", "1685638715"],
+        ["p", "npub1xtscya34g58tk0z605fvr788k263gsu6cy9x0mhnm87echrgufzsevkk5s"],
+        ["p", "npub13v47pg9dxjq96an8jfev9znhm0k7ntwtlh9y335paj9kyjsjpznqzzl3l8"],
+        ["p", "npub1zafcms4xya5ap9zr7xxr0jlrtrattwlesytn2s42030lzu0dwlzqpd26k5"],
```

Comment:
this test data looks wrong? they should be hex

Reply:
the model accounts for both hex and npubs

Reply:
This is wrong though, npubs are never at the protocol level

Reply:
Wait, these are still npubs. Was this intentional to test bad inputs or just missed changing these?